File: API/WorkoutTracker.Domain/Workouts/Workout.cs
using System;
using System.Collections.Generic;
using System.Text;
using WorkoutTracker.Domain.BaseClasses;
using WorkoutTracker.Domain.Exercises;
using WorkoutTracker.Domain.Interfaces;
using WorkoutTracker.Domain.Sets;

namespace WorkoutTracker.Domain.Workouts
{
    public class Workout : NamedEntity, IPublicEntity
    {
        public Guid PublicId { get; set; }
        public virtual ICollection<ExerciseInWorkout> Exercises { get; set; }
        public bool Active { get; set; }
    }
}

File: workout-tracker/src/app/workouts/models/workout.ts
import { NamedEntity } from "../../shared/models/named-entity";
import { ExerciseInWorkout } from './exercise-in-workout';

/*
THIS IS THE DOMAIN CLASS, NOT A DTO
*/

export class Workout extends NamedEntity {
  //public userId: number;
  public exercises: Array<ExerciseInWorkout>;
  public active: boolean;
  public publicId: string | null;

  constructor() {
    super();
    this.id = 0;
    this.exercises = [];
    this.active = true;
    //this.userId = 0;
    this.publicId = null;
  }
}
